src/frontend/components/form-group/autosave/lib/autosaveBase.ts

```diff
@@ -0,0 +1,36 @@
+import { Component } from "component";
+import gadsStorage from "util/gadsStorage";
+
+export default abstract class AutosaveBase extends Component {
+    constructor(element: HTMLElement) {
+       super(element);
+       this.initAutosave();
+    }
+
+    get isClone() {
+        return location.search.includes('from');
+    }
+
+    get layoutId() {
+        return $('body').data('layout-identifier');
+    }
+
+    get recordId() {
+        const id = location.pathname.split('/').pop();
```

Comment:
I think it should be possible to get the record ID from the body of the edit page (e.g. `data-current-id` in the form)

Reply:
I can't find this - only the curval has this.

Reply:
Is it here or have I misunderstood? https://github.com/ctrlo/GADS/blob/3e003fb1460d24c0e57fba986b55ba4eb7af4336/views/edit.tt#L22

Reply:
That's only on curvals.

I also didn't realise these comments are in realtime, I thought they were only when I did my push - that's useful to know! :smile:

Reply:
> That's only on curvals.

Fair enough, I hadn't spotted that, but maybe it could be moved outside of that condition? I haven't looked at it closely enough, but there should be something possible along those lines?